### PR TITLE
DFMS-229: addon uninstall is not removing all the resources

### PR DESCRIPTION
**Pull request by naveenpaul1**
addon uninstall is not removing all the resources: Fix

[DFMS-229](https://issues.redhat.com/browse/DFMS-229)

Signed-off-by: @naveenpaul1 Naveen Paul

### CODE REVIEW

File: controllers/managedmcg_controller.go [hunk 1]
```diff
@@ -436,6 +438,22 @@ func (r *ManagedMCGReconciler) removeNoobaa() error {
 			return fmt.Errorf("failed to delete bucketClasses: %w", err)
 		}
 	}
+	timeout := 10 * time.Second
```
Comment (v-harihar) [inline controllers/managedmcg_controller.go:441]: Is 10 secs enough ?

File: controllers/managedmcg_controller.go [hunk 2]
```diff
@@ -457,20 +475,6 @@ func (r *ManagedMCGReconciler) removeNoobaa() error {
 		}
 	}
 
-	objectBucketClaims := &noobaav1alpha1.ObjectBucketClaimList{}
```
Comment (v-harihar) [inline controllers/managedmcg_controller.go:460]: Why this is removed? Need your help to understand the patch

File: controllers/managedmcg_controller.go (hunk 1 above)
Comment (naveenpaul1) [reply controllers/managedmcg_controller.go:441]: yes @v-harihar, most of the time condition will be true within first iteration

File: controllers/managedmcg_controller.go (hunk 2 above)
Comment (naveenpaul1) [reply controllers/managedmcg_controller.go:460]: @v-harihar we added this code assuming that even if we remove the BucketClass OBC is not getting deleted in the uninstallation flow(we have code to delete OBC when respective BucketClass is deleted, bucketClassDeleted()). But the reason was mcg-osd CSV getting deleted without waiting for all the OBCs to delete. with the new PollImmediate() we are making sure all the OBCs and respective secret and Configmap are getting deleted.

**Pull request merged**